### PR TITLE
Update to 9.2 and enable python

### PR DESCRIPTION
**Pull request by agrayver**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

I wonder if starting from version 9.2 we could enable python bindings to be built? Something like what this PR suggests.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,11 +24,13 @@ requirements:
     - tbb-devel
     - muparser
     - boost-cpp
+    - python
```
Comment (xywei) [inline recipe/meta.yaml:27]: Seems that the CI job failed at build stage when `cmake` was trying to locate `boost.python`.

```suggestion
    - python
    - py-boost
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -24,11 +24,14 @@ requirements:
...
+    - boost
   run:
     - zlib
     - tbb
     - {{ pin_compatible("muparser") }}
     - boost-cpp
+    - python
 
 test:
```
Comment (xywei) [inline recipe/meta.yaml:36]:
```suggestion
test:
  imports:
    - PyDealII
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -37,6 +40,7 @@ test:
     - cmake
     - make
     - tbb-devel
+    - python
```
Comment (xywei) [inline recipe/meta.yaml:43]:
```suggestion
    - python
    - boost
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -24,11 +24,14 @@ requirements:
...
     - muparser
     - boost-cpp
+    - python
+    - boost
   run:
     - zlib
     - tbb
     - {{ pin_compatible("muparser") }}
     - boost-cpp
+    - python
```
Comment (xywei) [inline recipe/meta.yaml:34]:
```suggestion
    - python
    - boost
```

**Comment by xywei** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 5]
```diff
@@ -24,19 +24,27 @@ requirements:
     - tbb-devel
     - muparser
     - boost-cpp
```
Comment (xywei) [inline recipe/meta.yaml:26]: `boost-cpp` and `boost` seem to be conflicting
```suggestion

```

File: recipe/meta.yaml [hunk 6]
```diff
@@ -24,19 +24,27 @@ requirements:
...
+    - python
+    - boost
   run:
     - zlib
     - tbb
     - {{ pin_compatible("muparser") }}
     - boost-cpp
```
Comment (xywei) [inline recipe/meta.yaml:33]:
```suggestion
```

**Comment by xywei** [comment]
@conda-forge-admin, please rerender

**Comment by xywei** [comment]
On osx, the `cmake` script found `boost` but could not find `boost.python`
```
-- Found PythonLibs: $PREFIX/lib/libpython3.6m.dylib (found version "3.6.10") 
--   BOOST_VERSION: 1.70.0
--   BOOST_LIBRARIES: $PREFIX/lib/libboost_iostreams.dylib;$PREFIX/lib/libboost_serialization.dylib;$PREFIX/lib/libboost_system.dylib;$PREFIX/lib/libboost_thread.dylib;$PREFIX/lib/libboost_regex.dylib;$PREFIX/lib/libboost_chrono.dylib;$PREFIX/lib/libboost_date_time.dylib;$PREFIX/lib/libboost_atomic.dylib
--   BOOST_INCLUDE_DIRS: $PREFIX/include
--   BOOST_USER_INCLUDE_DIRS: $PREFIX/include
-- Found BOOST
CMake Error at contrib/python-bindings/CMakeLists.txt:47 (MESSAGE):
  DEAL_II_COMPONENT_PYTHON_BINDINGS has unmet configuration requirements: The
  external boost library does not provide Boost.Python
```

I have no idea how to fix this. Any ideas?

**Comment by agrayver** [comment]
Yes, this was fixed shortly after release, too bad... 

See https://github.com/dealii/dealii/pull/10380

@tamiko is there anything we can do here to make conda find boost on osx?

**Comment by isuruf** [comment]
You can add the fix as a patch. For eg: https://github.com/conda-forge/pyopencl-feedstock/blob/830a9751c4d9eea21ba9458663770286b5a8274d/recipe/meta.yaml#L14-L15

**Comment by agrayver** [comment]
I have added a patch, let's see.

**Comment by xywei** [comment]
It looks good to me.

@inducer Any additional thoughts? Is it okay with you to merge this?

**Comment by inducer** [comment]
I don't have much experience with deal.ii. This is your call.

File: recipe/meta.yaml [hunk 7]
```diff
@@ -23,20 +25,26 @@ requirements:
...
+  imports:
+    - PyDealII
   requires:
     - {{ compiler('cxx') }}
     - {{ compiler('fortran') }}
     - cmake
     - make
     - tbb-devel
+    - python
+    - boost
```
Comment (isuruf) [inline recipe/meta.yaml:47]: Above two lines should be deleted as they are run dependencies and should be pulled in automatically

Comment (agrayver) [reply recipe/meta.yaml]: I've removed these lines.

**Pull request merged**